src/Val.cc

```diff
@@ -3962,29 +3962,23 @@ ValManager::ValManager()
 
 	for ( auto i = 0u; i < PREALLOCATED_INTS; ++i )
 		ints[i] = Val::MakeInt(PREALLOCATED_INT_LOWEST + i);
-
-	for ( auto i = 0u; i < ports.size(); ++i )
-		{
-		auto& arr = ports[i];
-		auto port_type = (TransportProto)i;
-
-		for ( auto j = 0u; j < arr.size(); ++j )
-			arr[j] = IntrusivePtr{AdoptRef{}, new PortVal(PortVal::Mask(j, port_type))};
-		}
 	}
 
-const PortValPtr& ValManager::Port(uint32_t port_num, TransportProto port_type) const
+const PortValPtr& ValManager::Port(uint32_t port_num, TransportProto port_type)
 	{
 	if ( port_num >= 65536 )
 		{
 		reporter->Warning("bad port number %d", port_num);
 		port_num = 0;
 		}
 
-	return ports[port_type][port_num];
+	if ( ports.find(port_num) == ports.end() )
```

Comment:
I think `port_num` needs to be masked here and below based on:
```
uint32_t port_masked = PortVal::Mask(port_num, port_type)
...
return ports[port_masked];
```

Otherwise, for example, TCP port 53 and UDP port 53 will be confused and take the same spot in the map.

Reply:
@voidbar I fixed this on my fork, if you want to merge that PR in here.